File: sort/insertion_sort.c
void insertion_sort(int n, int v[]) {
    int i, j, valor;
    for (j = 1; j < n; j++) {
        valor = v[j];
        // Move os elementos maiores para a direita
        for (i = j - 1; i >= 0 && v[i] > valor; i--) {
            v[i + 1] = v[i];
        }
        // Insere o valor no lugar correto
        v[i + 1] = valor;
    }
}

void print_array(int n, int v[]) {
    for (int i = 0; i < n; i++) {
        printf("%d ", v[i]);
    }
    printf("\n");
}

int main() {
    int n = 20;
    int v[20] = {29, 10, 14, 37, 13, 5, 3, 1, 7, 25, 15, 11, 8, 19, 33, 20, 30, 22, 2, 17};

    printf("Vetor original: ");
    print_array(n, v);

    int i, j, valor;
    for (j = 1; j < n; j++) {
        valor = v[j];
        printf("\nInserindo o valor %d:\n", valor);
        
        for (i = j - 1; i >= 0 && v[i] > valor; i--) {
            v[i + 1] = v[i];
            printf("Deslocando: ");
            print_array(n, v);
        }
        
        v[i + 1] = valor;
        printf("Vetor atualizado com valor %d: ", valor);
        print_array(n, v);
    }

    printf("\nVetor ordenado final: ");
    print_array(n, v);

    return 0;
}

File: sort/insertion_sort.py
def insertion_sort(lista):
    for i in range(1, len(lista)):
        atual = lista[i]
        anterior = i - 1
        # Move os elementos da lista[0..i-1] que são maiores que o atual, uma posição à frente
        while anterior >= 0 and atual < lista[anterior]:
            lista[anterior + 1] = lista[anterior]
            anterior -= 1
        lista[anterior + 1] = atual

# Exemplo de uso
lista = [64, 25, 12, 22, 11]
print("Lista original:", lista)
insertion_sort(lista)
print("Lista ordenada:", lista)